### PR TITLE
New module: presto/filterseq

### PR DESCRIPTION
**Pull request by snafees**
Adding presto filterseq nf module along with test data

## PR checklist

Closes #XXX  

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: modules/nf-core/presto/filterseq/main.nf [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    tag "$meta.id"
+    label 'process_medium'
+
+    // TODO nf-core: List required Conda package(s).
+    //               Software MUST be pinned to channel (i.e. "bioconda"), version (i.e. "1.10").
+    //               For Conda, the build (i.e. "h9402c20_2") must be EXCLUDED to support installation on different operating systems.
+    // TODO nf-core: See section in main README for further information regarding finding and adding container addresses to the section below.
+    conda (params.enable_conda ? "bioconda::presto=0.7.1" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity:0.7.1--pyhdfd78af_0':
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/main.nf:28]:
```suggestion
        'https://depot.galaxyproject.org/singularity/presto:0.7.1--pyhdfd78af_0':
```

File: modules/nf-core/presto/filterseq/main.nf [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    // TODO nf-core: Where applicable please provide/convert compressed files as input/output
+    //               e.g. "*.fastq.gz" and NOT "*.fastq", "*.bam" and NOT "*.sam" etc.
+    tuple val(meta), path(R1), path(R2)
+
+    output:
+    tuple val(meta), path("*R1_quality-pass.fastq"), path("*R2_quality-pass.fastq") ,  emit: reads
+    path "*_command_log.txt" , emit: logs
+    path "versions.yml" , emit: versions
+    path "*_R1.log" , emit: R1_log
+    path "*_R2.log" , emit: R2_log
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/main.nf:45]:
```suggestion
```

File: modules/nf-core/presto/filterseq/meta.yml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+      description: File containing software versions
+      pattern: "versions.yml"
+  - reads:
+      type: file
+      description: R1 & R2 quality passed fastq files"
+      pattern: "*.fastq"
+  - logs:
+      type: files
+      escription: File containing log information
+      pattern: "*.log"
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/meta.yml:53]:
```suggestion
```

**Review by ggabernet — COMMENTED** [review]
Hi @snafees , thanks a lot for your contribution! I would just have some comments :D

File: modules/nf-core/presto/filterseq/main.nf [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    // TODO nf-core: Where possible, a command MUST be provided to obtain the version number of the software e.g. 1.10
+    //               If the software is unable to output a version number on the command-line then it can be manually specified
+    //               e.g. https://github.com/nf-core/modules/blob/master/modules/homer/annotatepeaks/main.nf
+    //               Each software used MUST provide the software name and version number in the YAML version file (versions.yml)
+    // TODO nf-core: It MUST be possible to pass additional parameters to the tool as a command-line string via the "task.ext.args" directive
+    // TODO nf-core: If the tool supports multi-threading then you MUST provide the appropriate parameter
+    //               using the Nextflow "task" variable e.g. "--threads $task.cpus"
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/main.nf:58]:
```suggestion
```

File: modules/nf-core/presto/filterseq/main.nf [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    // TODO nf-core: Please replace the example samtools command below with your module's command
+    // TODO nf-core: Please indent the command appropriately (4 spaces!!) to help with readability ;)
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/main.nf:60]:
```suggestion
```

File: modules/nf-core/presto/filterseq/meta.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
+name: "presto_filterseq"
+## TODO nf-core: Add a description of the module and list keywords
+description: Filtering sequences based on quality.
+keywords:
+  - presto
+  - FilterSeq
+tools:
+  - "presto":
+      ## TODO nf-core: Add a description and other details for the software below
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/meta.yml:9]:
```suggestion
```

File: modules/nf-core/presto/filterseq/meta.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+      description: "A bioinformatics toolkit for processing high-throughput lymphocyte receptor sequencing data."
+      homepage: "https://presto.readthedocs.io/en/stable/install.html"
+      documentation: "https://presto.readthedocs.io/en/stable/"
+      tool_dev_url: "https://bitbucket.org/kleinstein/presto/src/master/"
+      doi: "10.1093/bioinformatics/btu138"
+      licence: "['AGPL v3']"
+
+## TODO nf-core: Add a description of all of the variables used as input
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/meta.yml:17]:
```suggestion
```

File: modules/nf-core/presto/filterseq/meta.yml [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+  - R1:
+      type: file
+      description: fastq read pair 1
+      pattern: "*.fastq"
+  - R2:
+      type: file
+      description: fastq read pair 2
+      pattern: "*.fastq"
+
+## TODO nf-core: Add a description of all of the variables used as output
```
Comment (ggabernet) [inline modules/nf-core/presto/filterseq/meta.yml:34]:
```suggestion
```

**Review by ggabernet — COMMENTED** [review]
And just removing all TODOs :D

**Comment by ggabernet** [comment]
@nf-core-bot fix linting

**Review by ggabernet — APPROVED** [review]
Looks good to me now!

**Comment by ggabernet** [comment]
@nf-core-bot fix linting

File: modules/nf-core/presto/filterseq/main.nf [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+    FilterSeq.py quality -s $R1 $args --outname "${prefix}_R1" --log "${R1.baseName}_R1.log" --nproc ${task.cpus} > "${meta.id}_command_log.txt"
+    FilterSeq.py quality -s $R2 $args --outname "${prefix}_R2" --log "${R2.baseName}_R2.log" --nproc ${task.cpus} >> "${meta.id}_command_log.txt"
+    ParseLog.py -l "${R1.baseName}_R1.log" "${R2.baseName}_R2.log" -f ID QUALITY
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        presto: \$( FilterSeq.py --version | awk -F' '  '{print \$2}' )
+    END_VERSIONS
+    """
+}
```
Comment (nvnieuwk) [inline modules/nf-core/presto/filterseq/main.nf:35]: Can you add a stub? :)

**Comment by ggabernet** [comment]
Closing this PR as I am working on it now with the new modules template with nf-test

**Pull request closed without merging**